src/routes/transactions/transactions.controller.spec.ts

```diff
@@ -0,0 +1,271 @@
+import { faker } from '@faker-js/faker';
+import { INestApplication } from '@nestjs/common';
+import { Test, TestingModule } from '@nestjs/testing';
+import { readFileSync } from 'fs';
+import * as request from 'supertest';
+import {
+  fakeConfigurationService,
+  TestConfigurationModule,
+} from '../../config/__tests__/test.configuration.module';
+import {
+  fakeCacheService,
+  TestCacheModule,
+} from '../../datasources/cache/__tests__/test.cache.module';
+import {
+  mockNetworkService,
+  TestNetworkModule,
+} from '../../datasources/network/__tests__/test.network.module';
+import { DomainModule } from '../../domain.module';
+import chainFactory from '../../domain/chains/entities/__tests__/chain.factory';
+import { DataSourceErrorFilter } from '../common/filters/data-source-error.filter';
+import { TransactionsModule } from './transactions.module';
+
+describe('Transactions Controller (Unit)', () => {
+  let app: INestApplication;
+  let safeConfigApiUrl: string;
+
+  beforeAll(async () => {
+    safeConfigApiUrl = faker.internet.url();
+    fakeConfigurationService.set('safeConfig.baseUri', safeConfigApiUrl);
+    fakeConfigurationService.set('exchange.baseUri', faker.internet.url());
+    fakeConfigurationService.set('exchange.apiKey', faker.datatype.uuid());
+  });
+
+  beforeEach(async () => {
+    jest.clearAllMocks();
+    fakeCacheService.clear();
+
+    const moduleFixture: TestingModule = await Test.createTestingModule({
+      imports: [
+        // feature
+        TransactionsModule,
+        // common
+        DomainModule,
+        TestCacheModule,
+        TestConfigurationModule,
+        TestNetworkModule,
+      ],
+    }).compile();
+
+    app = moduleFixture.createNestApplication();
+    app.useGlobalFilters(new DataSourceErrorFilter());
+
+    await app.init();
```

Comment:
Do we need to init the app before running each test or could it work with only doing it once? When I move this part into `beforeAll`, some tests randomly fail with a 503 error. In the [nestjs documentation](https://docs.nestjs.com/fundamentals/testing) they are also calling `init` in `beforeAll`.

Reply:
Yes, good catch, I also tried to init the app once in the `beforeAll` hook, and noted that. I'm not sure why this is happening, maybe it's something regarding the app configuration. I didn't spend so much time on it honestly but could be interesting to tackle it in a separate issue since it's not specific to this PR. Thanks!